.github/workflows/asciidoctor-ghpages.yaml

```diff
@@ -1,7 +1,10 @@
 name: GitHub Pages Publish
 on:
-  push:
-    branches: [main]
+  workflow_run:
+    workflows: [Main Push Trigger]
+    types: [completed]
+concurrency:
+  group: ${{ github.workflow }}
```

Comment:
We do not specify which ref to build and deploy so the checkout action checks out the default branch (main). This change would prevent jobs racing to update the ghpages branch. Practically I don't think this can happen given the need to run 10 minutes of builds after rebasing onto main to get your branch mergable, but lets be paranoid.

If preferred I imagine we could dig out the ref used by `Main Push Trigger` from the `workflow_run`.

Reply:
okay I've changed it to use the head commit id from the Main Push Trigger build, seemed nicer to tie it to the triggering build

Reply:
Is there a change missing from this branch? I still see the concurrency as 
```yaml
concurrency:
  group: ${{ github.workflow }}
```
But your comment implies it should be 

```yaml
concurrency:
  group: ${{ github.event.workflow_run.head_commit.id}}
```
or similar...

Reply:
the concurrency group is limiting it to a single concurrency group for `GitHub Pages Publish` so that we don't get parallel runs of this action. They should be queued up (I'll check that it doesn't cancel jobs by default).

I'd still worry that it's theoretically possible for them to clobber each other if you somehow managed to push multiple times to main in quick succession and let parallel runs of this action occur. Each would have the right ref but there would be a race to push to the `ghpages` branch.

Reply:
>  When a concurrent job or workflow is queued, if another job or workflow using the same concurrency group in the repository is in progress, the queued job or workflow will be pending. Any pending job or workflow in the concurrency group will be canceled. This means that there can be at most one running and one pending job in a concurrency group at any time.

I guess that works? It means if we had 1 build running and multiple queued up it would cancel the older pending job.

Reply:
it's a bit moot since we need a green build to get into main, so we shouldn't have rapid consecutive merges, but worth being a little paranoid

Reply:
sorry I've still been unclear, there's two things in this thread:

1. limiting concurrency to prevent races to update the ghpages branch
2. checking out the reference from the triggering build, this is more in the spirit of reproducibility, it's not strictly required.